internal/notif/ably_notif.go

```diff
@@ -24,27 +26,53 @@ type PushNotificationBody struct {
 	Body  string `json:"body"`
 }
 
-type PushNotificationRecipient struct {
+type ApnsPushNotificationRecipient struct {
 	TransportType transportType `json:"transportType"`
 	DeviceToken   string        `json:"deviceToken"`
 }
 
+type FcmPushNotificationRecipient struct {
+	TransportType     transportType `json:"transportType"`
+	RegistrationToken string        `json:"registrationToken"`
+}
+
 type ApplePushNotification struct {
-	Recipient    PushNotificationRecipient `json:"recipient"`
-	Notification PushNotificationBody      `json:"notification"`
+	Recipient    ApnsPushNotificationRecipient `json:"recipient"`
+	Notification PushNotificationBody          `json:"notification"`
+}
+
+type AndroidPushNotification struct {
+	Recipient    FcmPushNotificationRecipient `json:"recipient"`
+	Notification PushNotificationBody         `json:"notification"`
 }
 
 func SendPushNotification(ctx context.Context, config config.AblyConfig, device *model.UserDevice, pushBody PushNotificationBody) (bool, error) {
 	if !config.Enabled {
 		return true, nil
 	}
-	body, err := json.Marshal(ApplePushNotification{
-		Recipient: PushNotificationRecipient{
-			TransportType: "apns",
-			DeviceToken:   *device.Token,
-		},
-		Notification: pushBody,
-	})
+
+	var body []byte = nil
+	var err error = nil
+	if strings.ToLower(device.Platform) == "android" {
```

Comment:
Should we do a verification that `device.Token` is not `nil`?

Reply:
Yeah that’s true. I’ll add it in the next commit.